### PR TITLE
Get 418 to work

### PR DESCRIPTION
**Pull request by ksabbak**
You know when you take forever to write an email and then you expect the email to be better and better the longer it takes you to send it in order to justify the length of time it took you to get back to the person, but the email does not improve and it is still just an acceptable email and you feel like you should have sent it sooner but it's too late now? That's how I feel about this particular PR.

Also, suggestions on what to do with the switch/conditionals that are likely going to keep multiplying would be appreciated.

### CODE REVIEW

File: src/main/java/com/ksabbak/javaserver/app/controller/ControllerFactory.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        switch(path) {
+            case "/coffee":
+                return new CoffeeController(methods);
+            default:
+                return new Controller(methods);
+        }
+    }
+}
+
+//public String getTypeOfDayWithSwitchStatement(String dayOfWeekArg) {
```
Comment (BWoodfork) [inline src/main/java/com/ksabbak/javaserver/app/controller/ControllerFactory.java]: Dead code

Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/controller/ControllerFactory.java]: Whoops, thanks!

File: src/main/java/com/ksabbak/javaserver/app/Responder.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    private Controller controller;
+
+    public Responder(Controller controller){
+        this.controller = controller;
+        Object[] methods = controller.getMethods();
+        if (Arrays.asList(methods).contains(HTTPMethod.GET)){
+            setGetStatus();
+        }
+        if (Arrays.asList(methods).contains(HTTPMethod.GET)){
```
Comment (hnlee) [inline src/main/java/com/ksabbak/javaserver/app/Responder.java]: Should this be `HTTPMethod.HEAD`?

File: src/main/java/com/ksabbak/javaserver/app/Responder.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        return statusHead;
+    }
+
+    private void setHeadStatus(){
+        statusHead = StatusCode.OK;
+    }
+
+    public String getGetBody(){
+        return controller.bodyGet();
+    }
```
Comment (hnlee) [inline src/main/java/com/ksabbak/javaserver/app/Responder.java:65]: I found this a bit hard to follow.

What about if the controller returned an object that contained both a status and a body?

File: src/main/java/com/ksabbak/javaserver/app/Routes.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Routes {
+    private static final Map<String, Object[]> routes = initMap();
+
+    private static Map<String, Object[]> initMap() {
+        Map<String, Object[]> map = new HashMap<String, Object[]>();
+        map.put("/", new Object[]{HTTPMethod.GET, HTTPMethod.HEAD});
```
Comment (hnlee) [inline src/main/java/com/ksabbak/javaserver/app/Routes.java]: I wonder if it would be easier if you had a separate route for every method and path combination. You might not want GET / to return the same response as HEAD /.

File: src/main/java/com/ksabbak/javaserver/app/Responder.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+public class Responder {
+    protected StatusCode statusGet = StatusCode.NOT_FOUND;
+    protected StatusCode statusHead = StatusCode.NOT_FOUND;
+
+    private Controller controller;
+
+    public Responder(Controller controller){
+        this.controller = controller;
+        Object[] methods = controller.getMethods();
+        if (Arrays.asList(methods).contains(HTTPMethod.GET)){
```
Comment (hnlee) [inline src/main/java/com/ksabbak/javaserver/app/Responder.java]: Maybe store them as a `List` to begin with? Is there an advantage to storing as an `Array`?

File: src/main/java/com/ksabbak/javaserver/app/Routes.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import com.ksabbak.javaserver.server.HTTPMethod;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Routes {
+    private static final Map<String, Object[]> routes = initMap();
+
+    private static Map<String, Object[]> initMap() {
```
Comment (hnlee) [inline src/main/java/com/ksabbak/javaserver/app/Routes.java]: Why is it a generic `Object[]` instead of an `HTTPMethod[]`?

**Review by hnlee — CHANGES_REQUESTED** [review]
You don't need to refactor right away on the comment about my finding the `Responder` hard to follow or the comment on the routes, but keep them in mind as the design of your server evolves.

Other than that `Responder` class, it's very clean and readable, and the unit tests are looking good!

File: src/main/java/com/ksabbak/javaserver/app/Responder.java (hunk 2 above)
Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/Responder.java]: Yes it should! Thanks for catching that.

File: src/main/java/com/ksabbak/javaserver/app/Responder.java (hunk 3 above)
Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/Responder.java:65]: Realized today that that's where this is headed, that refactor will happen when I get to Simple OPTION. I'm glad that will also increase clarity as well as like, make things work better.

File: src/main/java/com/ksabbak/javaserver/app/Routes.java (hunk 4 above)
Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/Routes.java]: I'd love to pick your brain about this a bit more next week. Can we set aside some time to chat?

File: src/main/java/com/ksabbak/javaserver/app/Responder.java (hunk 5 above)
Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/Responder.java]: Probably not. I'll double check, but I think it was just a Java fail. >.>

File: src/main/java/com/ksabbak/javaserver/app/Routes.java (hunk 6 above)
Comment (ksabbak) [reply src/main/java/com/ksabbak/javaserver/app/Routes.java]: Another Java fail! I'll fix this too. Thanks!

File: src/main/java/com/ksabbak/javaserver/app/Routes.java (hunk 4 above)
Comment (hnlee) [reply src/main/java/com/ksabbak/javaserver/app/Routes.java]: Sure, let's set up a time to either pair or do a Slack call.
